core/ArchiveProcessor/Rules.php

```diff
@@ -343,4 +344,53 @@ public static function shouldProcessSegmentsWhenReArchivingReports()
     {
         return Config::getInstance()->General['rearchive_reports_in_past_exclude_segments'] != 1;
     }
+
+    public static function shouldForceArchiveInvalidatedArchiveInThisRequest(Parameters $params, $doneFlags)
+    {
+        $isArchivePhpTrigerred = !SettingsServer::isArchivePhpTriggered(); // safety to make sure core:archive doesn't trigger archiving if the archive is still usable
+        if (!$isArchivePhpTrigerred) {
+            return false;
+        }
+
+        $isArchivingEnabledFor = !Rules::isArchivingDisabledFor([$params->getSite()->getId()], $params->getSegment(), $params->getPeriod()->getLabel());
+        if (!$isArchivingEnabledFor) {
+            return false;
+        }
+
+        // if the period includes today, then we only rearchive if the archive is past the TTL (to avoid rearchiving the range
+        // every time it's viewed on a high traffic site)
+        $periodIncludesToday = $params->getPeriod()->isDateInPeriod(Date::factory('today'));
+        if ($periodIncludesToday) {
+            return false;
+        }
+
+        // in this case, the archive is a segment that is not pre-processed. for these, if archiving is enabled for this request
+        // then child archiving will definitely be allowed, so we can just return true here.
```

Comment:
how do we know that in this case it's a segment that is not pre-processed? And does this consider `browser_archiving_disabled_enforce`?